// `panic` は通常、何かが予期せず誤った結果になったことを意味します。
// ほとんどの場合、通常の操作では起こるはずがない、すなわち、
// うまく扱えないエラーが起きたときに、異常終了させるために使います。

package main

import "os"

func main() {

    // 予期せぬエラーを確認するために、panic を使います。
    // これは、panic するために作られた唯一のプログラムです。
    panic("a problem")

    // `panic` の一般的な使い方は、ある関数が扱い方を知らない
    // (または、扱いたくない) エラー値を返したときに、
    // 異常終了させることです。
    // 新規ファイル作成時に予期せぬエラーが発生したら
    // `panic` する例は、次の通りです。
    _, err := os.Create("/tmp/file")
    if err != nil {
        panic(err)
    }
}
